### DISCUSSION TITLE
remote access to logisim-evolution?

### DISCUSSION THREAD

**Discussion by rm144** [logisim-evolution/logisim-evolution#1839] [Q&A] [answered]
When I try to start logisim-evolution on a remote machine (Ubuntu, both on local and remote machine, remote login via `ssh -X`), I get the following error:

```
% logisim-evolution
Picked up _JAVA_OPTIONS: -Duser.home=/home/myname/snap/logisim-evolution/4
Exception in thread "main" java.awt.AWTError: Can't connect to X11 window server using 'localhost:10.0' as the value of the DISPLAY variable. 
```
whereas other programs with graphical interface running on the remote machine can display their windows on the local machine.

This seems to be a Java-related problem, and https://stackoverflow.com/a/11525353/3852630 suggests to pass the option
`-Djava.awt.headless=true` to Java. All the other X11-related solutions (setting `DISPLAY`, `xhost +` etc.) didn't work for me.

Does anyone know how I can start logisim-evolution on a remote Linux machine or how this Java option can be passed to logisim-evolution?

It would be nice if our students could work on the machines in our lab from remote.

Thanks for your help!

Ralf

**Comment by maehne** [comment]
You seem to be using Logisim-evolution packaged as a Snap. Could you please test using the native DEB package and JAR archive ([version 3.8.0](https://github.com/logisim-evolution/logisim-evolution/releases/tag/v3.8.0) and [nightly build](https://github.com/logisim-evolution/logisim-evolution/actions/runs/6018814906)), whether they show also this behaviour?

**Reply by rm144** [reply]
Thanks. Yes, the snap installation is on my laptop. We have other machines where it was installed by our sysops from flatpak (https://flathub.org/apps/com.github.reds.LogisimEvolution). Is this similar to the DEB package? Then I would try this one (as soon as I get access).

**Reply by maehne** [reply]
Flatpaks are more similar to Snap packages than to a DEB package, as both execute the software in a sandbox. In general, it is preferable to use the native package format of your distribution - especially if you would like to use advanced features such as the possibility to download your designs to FPGA or co-simulate with ModelSim/QuestaSim. Nevertheless, a test of the Flatpak is also worth.

**Reply by rm144** [reply]
So it could be a problem with the sandbox?

**Reply by maehne** [reply]
Yes.

**Reply by rm144** [reply]
Ah, I see, thanks a lot! I can report how to start it from remote when it is installed via flatpak: `flatpak --share=network run com.github.reds.LogisimEvolution`. Without the `--share=network`, I get the same error as described in my question. So I have to figure out how to do the same with snap.

**Reply by rm144** [reply, marked as answer]
I did not succeed for snap (none of the solutions from https://forum.snapcraft.io/t/x11-forwarding-using-ssh/2381 and https://stackoverflow.com/questions/61933724/snap-applications-through-a-remote-x11-session helped). So if anyone has an idea, I'd be grateful. But I can live with the flatpak solution for now and maybe abandon the snap solution on the laptop. Thanks a lot for your help!

**Reply by maehne** [reply]
So executing `export XAUTHORITY=$HOME/.Xauthority` on the remote host prior launching Logisim-evolution from the same shell did not help?

As it seems to be a Snap limitation rather a problem in our packaging, I won't create an issue for the moment. Nevertheless, it would be nice to post here if you should find any solution.

**Reply by rm144** [reply]
No, unfortunately setting `XAUTHORITY` didn't work for me, and also creating the symbolic link from the snap directory to `$HOME/.Xauthority` didn't help.

Do you happen to know how I could pass `-Djava.awt.headless=true` (maybe in addition) to logisim-evolution? But maybe that's difficult, because `logisim-evolution` is actually a link to `/usr/bin/snap` and not the binary of `logisim-evolution`, and then there's also Java involved at some point.

**Reply by maehne** [reply]
The option `-Djava.awt.headless=true` needs to be added to [line 47 of `snapcraft.yaml`](https://github.com/logisim-evolution/logisim-evolution/blob/eda29b4a01bf6a2d78c86e980e9d8fd949726d3c/snap/snapcraft.yaml#L47) and then you need to [build the Snap](https://snapcraft.io/docs/java-applications).

**Reply by rm144** [reply]
Thanks a lot! I'll need to find some quiet moment to test this.
